metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -450,8 +475,36 @@ def _find_measure_recipe(
                 for y in x.join_on_partition_time_dimensions
             )
 
+            measure_node = node_with_lowest_cost
+            # Since every build_aggregate_measures function call groups measure_specs by additiveness
+            # all of these measure_specs will have the same non_additive_dimension_spec
+            non_additive_dimension_spec: Optional[NonAdditiveDimensionSpec] = measure_specs[
+                0
+            ].non_additive_dimension_spec
+            if non_additive_dimension_spec is not None:
+                agg_time_dimension = self._data_source_semantics.get_agg_time_dimension_for_measure(
+                    measure_specs[0].as_reference
+                )
```

Comment:
These are still unchecked. If you look on line 349 we have an invariant check enforcing the internal contract for `_find_measure_recipe`. I think we need to extend that check, because we are now relying on this extended set of conditions (i.e., there is only one agg time dimension and only one non-additive spec) internally.

You can update the contract in one of two ways - either do it via a runtime assertion, or update the signature to accept a non-additive dimension spec and an agg time dimension. The former is simplest and probably cleanest, especially given that this is an internal helper method. The latter is more ergonomic as the signature itself tells developers what to expect.

The best thing might be to overhaul the signature to take in an object representing the measure specs, with properties for data source, agg time dimension, and non-additive dimension. That object can then maintain the invariants and encapsulate the work of doing the extraction of these values at the callsites. But it's more work.

Reply:
I added the invariant checks with a helper function that takes in a `Sequence[MeasureSpec]` and it asserts that all those measure_specs has the same `agg_time_dimension` and `non_additive_dimension_spec`